### ISSUE TITLE
Release R1 Translator model on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [Mmurrad/EEG-To-text#1]
Hello,

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability.If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim
the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

It'd be great to make the checkpoints and dataset available on the 🤗 hub, to improve their discoverability/visibility.
We can add tags so that people find them when filtering https://huggingface.co/models and https://huggingface.co/datasets.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`. Alternatively, one can leverages the [hf_hub_download](https://huggingface.co/docs/huggingface_hub/en/guides/download#download-a-single-file) one-liner to download a checkpoint from the hub.

We encourage researchers to push each model checkpoint to a separate model repository, so that things like download stats also work. We can then also link the checkpoints to the paper page.

## Uploading dataset

Would be awesome to make the dataset available on 🤗 , so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the first few rows of the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗

**Comment by justin-xzliu** [comment]
Hi @NielsRogge and HF team🤗 — uploading datasets and checkpoints to the Hub for better visibility is a great idea!

We’ve recently released [GLIM](https://github.com/justin-xzliu/GLIM), another open-source EEG-to-text model built with PyTorch Lightning and HF Transformers. It's designed with strict data splits and quantitative semantic evaluations to improve reliability, and has been gaining increasing attention over the past month.

Although our code is organized for easy reproducibility, we haven’t yet taken advantage of the 🤗Hub for sharing the model and dataset. So we’d like to ask:
- Is GLIM’s current checkpoint (exported by Lightning, without HF-integrated modules) compatible with the Hub?
- Would publishing require major refactoring?
- Also, since EEG-to-text is an emerging area, would it be possible to add a new task tag like `brain-to-text` or `eeg-to-text` under the multimodal category on the Hub?

We believe these efforts could help improve discoverability and encourage more transparent benchmarking as the field continues to grow.

Thanks again for your great work — we’d be happy to contribute further!

**Comment by NielsRogge** [comment]
Very cool!

I see that the GLIM model checkpoint is available here: https://figshare.com/articles/preprint/glim-zuco-epoch_199-step_49600_ckpt/29115161/1?file=54689960, would be great to make it available on the hub instead. I've indexed the paper for it here: https://huggingface.co/papers/2505.17099.

Yes any model is compatible with the hub. As for integration, there are a couple of options:

## 1. Mixin class (least work)

In case it fits we could add some HF utility classes to allow pushing/loading to and from the hub. It could be the following works out-of-the-box (ensuring `pip install --upgrade huggingface_hub` has been ran locally):

```diff
+ from huggingface_hub import PyTorchModelHubMixin

class GLIM(L.LightningModule, 
+          PyTorchModelHubMixin,
+          repo_url="https://github.com/justin-xzliu/GLIM",
+          paper_url="https://huggingface.co/papers/2505.17099",
+          license="mit",
+          tags=["brain-to-text", "eeg-to-text"],
):
```

By just inheriting, the following works:
```python
from xxx import GLIM

model = GLIM(...)

# equip with weights
model.load_state_dict(...)

# push it to the hub
model.push_to_hub("your-hf-org-or-username/glim")

# now anyone can use it like so
model = GLIM.from_pretrained("your-hf-org-or-username/glim")
```

## 2. add helper functions

If you prefer to explicitly define pushing and loading yourself, then I recommend adding a minimal from_pretrained method to the class as explained here: https://huggingface.co/docs/huggingface_hub/guides/integrations#a-flexible-approach-helpers
